presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -244,7 +244,7 @@ default void beginQuery(ConnectorSession session)
     {}
 
     /**
-     * Notification of ending a SELECT/UPDATE/INSERT/DELETE query
+     * Notification of ending a query
```

Comment:
above should be "Start a query" also

Probably should add the bit about it being triggered for both failure and success here too